lib/OpenQA/WebAPI/Controller/Admin/User.pm

```diff
@@ -44,7 +48,13 @@ sub update {
         $self->flash('error', "Can't find that user");
     }
     else {
-        $user->update({is_admin => $is_admin, is_operator => $is_operator});
+        my $data = {is_admin => $is_admin, is_operator => $is_operator};
+        $data->{username} = $username if (defined($username));
+        $data->{email} = $email if (defined($email));
+        $data->{fullname} = $fullname if (defined($fullname));
+        $data->{nickname} = $nickname if (defined($nickname));
+
```

Comment:
that should be equivalent to just using the parameters regardless if they are define. Also it's a bit lengthy to assign first from `$self->param(…)` to internal variables and then the hash. Maybe it's ok to simply update the user completely from all specified params, something like `$user->update($self->req->params->to_hash)`

Reply:
We should do a proper validation here.

Simply forwarding the hash like *might be a security nightmare* because it would allow to set arbitrary columns.

I'm also wondering whether it makes sense to make all these columns editable. Some values are uses to identify the user within the underlying authentication provider and it should likely not be modified.